### PR TITLE
Implement JetSetllly and scorpheus improvements

### PR DESCRIPTION
**Pull request by neclepsio**
- GetScroll* and SetScroll*
- drag and drop
- color styles
- ContentRegionMax
- CollapsingHeader
- BeginPopup
- ItemSpacing
- two custom widgets, Spinner and Splitter, in Utils* files.

### CODE REVIEW

**Comment by neclepsio** [comment]
Sorry, it was premature. For example, CollapsingHeader signature is obsolete. I will resubmit it afet some testing.

File: imgui.go [hunk 1]
```diff
@@ -1227,6 +1279,56 @@ func ColumnsCount() int {
...
+// When this returns true you need to: a) call SetDragDropPayload() exactly once, b) you may render the payload visual/description, c) call EndDragDropSource()
+func BeginDragDropSource(flags int) bool {
+	return C.iggBeginDragDropSource(C.int(flags)) != 0
+}
+
+// SetDragDropPayload sets the payload for current draw and drop source.
+// type is a user defined string of maximum 32 characters.
+// Strings starting with '_' are reserved for dear imgui internal types.
+// Data is copied and held by imgui.
+func SetDragDropPayload(type_ string, payload string, cond Condition) bool {
```
Comment (dertseha) [inline imgui.go:1293]: In the library, the payload (called `data` in the API by the way) is handled rather as an arbitrary byte array. It should be exposed as such here as well.

File: imgui.go [hunk 2]
```diff
@@ -1227,6 +1279,56 @@ func ColumnsCount() int {
...
+	typeArg, typeFin := wrapString(type_)
+	defer typeFin()
+	out := payload + "\000" + strings.Repeat(" ", 512-len(payload))
```
Comment (dertseha) [inline imgui.go:1296]: This is extra magic that misses explanation and I suspect this is type-specific. If the `data` is handled as arbitrary byte array, this extra magic goes to the caller.
This also includes the constant of 512 as the internal argument. Please rework.

File: imgui.go [hunk 3]
```diff
@@ -1227,6 +1279,56 @@ func ColumnsCount() int {
...
+
+// BeginDragDropTarget must be called after submitting an item that may receive an item.
+// If this returns true, you can call AcceptDragDropPayload() and EndDragDropTarget().
+func BeginDragDropTarget() bool {
+	return C.iggBeginDragDropTarget() != 0
+}
+
+// AcceptDragDropPayload accepts contents of a given type.
+// If ImGuiDragDropFlags_AcceptBeforeDelivery is set you can peek into the payload before the mouse button is released.
+func AcceptDragDropPayload(type_ string, flags int) string {
```
Comment (dertseha) [inline imgui.go:1316]: Rework this function as well, please. Return type []byte and the internal functions need reworking for this.
(This then also removes the hardcoded 512).

**Comment by neclepsio** [comment]
Ok, as I wrote in the description I simply merged scorpheus improvements. I just started using ImGui and your wrapper. so please excuse if something I make big errors. As for comments as proper sentences, I read the contributing guidelines, and rewrote many of the comments, but missed some. 

P.S.: would you accept a refactoring that, for example, moved the `Column*` functions to new .go/.cpp/.h files? I noticed that the majority of functions are in `imgui.go` but, for example, you separated some in `Style.go`.

**Comment by dertseha** [comment]
All good - I appreciate your effort to collect some extensions that others made and didn't propose.
When adding features upstream, then additions need more consideration than were perhaps necessary for a specific fork.
You can still add some of the extensions if you don't want to spend extra effort on all the fixes.

It may be useful to extract some feature-blocks into separate files, yes. `imgui.go` started as the general bin for all the "global" functions, yet this does not scale well.
Unless it's a minor change, I recommend doing such refactorings as dedicated pull requests.

**Pull request closed without merging**